sources/pmwebapi.go

```diff
@@ -0,0 +1,164 @@
+package sources
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type GetToken struct {
+	Context int
+}
+
+type MetricHeader struct {
+	Metrics []PcpMetrics
+}
+
+type PcpMetrics struct {
+	Name        string
+	Pmid        int64
+	Indom       int64
+	TextOneline string `json:"text-oneline"`
+	TextHelp    string `json:"text-help"`
+	Sem         string
+	Units       string
+	Type        string
+}
+
+type TimestampHeader struct {
+	Timestamp
+	Values []GetValues
+}
+
+type Timestamp struct {
+	S  string
```

Comment:
I like this suggestion, though would be ok with it being a separate follow-on PR.